readme.md
![](anet_logo.png)
# ANet BBS - Curent Release: pre Alpha v.01A

This is the start of a replacement for CNet Pro BBS for the Amiga. With the lack of support or information with CNet, and the lack of classic hardware and emulation solutions limiting some of the features of the software, a replacement is needed. Something that will run on modern machines and still give you the great experience that Sysops and users could get with CNet on the Commodores and the Amigas. 

ANet is meant to be that solution, running on linux first and then to be ported to AROS and possibly AmigaOS, MorphOS and Mac. Primary focus will be on Linux as AROS makes use of the linux kernel and drivers. This can help speed the development along. 

Some of the features that ANet should have, and are also features in CNet:


   * A threaded message system. Just about all the other BBS packages don't support this. Or they have a system in place but don't display them in a threaded format. It's just WRONG!
   * Built in FTN support. Using JAMlib to import and export messages, it should be built in as well as have it's own tosser and scanner. 
   * Easy expansion for FTP, NNTP, Web and more. 
   * Sticky posts - SubOps can make some posts sticky for like rules or important announcements that don't need to go into the news section.
   * Rate posts - Users can rate on a scale of 1-5 stars on a post or it's replies. 

More will undoubtedly be added to the above list as it matures and progresses. Progress should be done in stages to get the system as stable as possible for release. The package will be open source so the BBS community at large can help make it better. 


## Files
———————————————————————————

bbscore.py - The main code of the BBS. Running this from a terminal will launch you into the BBS. Using a modified telnet daemon, allows access from the outside. Eventually, a whole socket routine will be written so it cal all be self contained. 

bbstext.txt - This will be the file bbscore will read from for it’s prompts. This is so it is easier to translate the board into other languages. 

bbsmenu.txt - This is the file for the menus of the BBS that will allow aliasing of commands. Currently, all commands are hard coded into the ‘main()’ function. 

modem_test.zip - A test file used by the z-modem file transfer function to test sending a single file. 

## Folders
———————————————————

users - Where all the user files will be kept. Currently it’s all just plain text, but will be switched over to a sqlite database in the future. 

MCI - multiple versions of code for an MCI interpreter language for the BBS. See the readme.txt in the folder for more info. 

news - Used for keeping news announcements for the BBS. A simple date checking is enabled, but the core BBS doesn’t log calls yet, which will be added in a future version of the alpha code of bbscore. 

news/headers - a separate folder where ANSI headers are stored that the news code will display before the news text file. 

systest - System text files are stores here, such as the welcome screen and logout screen. 

sysdata - System data files will go here. A complete list will be coming soon as the older CNet files we are using as templates are brought over. 

mteld14 - modified telnet daemon that can be used to launch the BBS on a telnet connect. 


### Release Notes
------------------------------------
Dec 1st, 2013 : pre Alpha v.01A
- Some of the "basics" are working. Users can login, get 3 tries for correct password, pressing ENTER 3 times gets you a terse message to contact SYSOP.
- Simple news display is working. Will display all news items since last call at login. But last call date is not being saved yet. That comes with logging. Will add a menu system for reading news next.
- Main Menu is still hard coded in. Some basic commands will be put in till a BBSMENU file system is fleshed out. Have some ideas, but nothing coded yet.
- User data files are still just txt files. Makes it easier to edit and test user related features. When things get further along, we'll go to sqlite db.
- New user signup is still not going to happen till I have more of the BBS fleshed out and able to accept incoming calls. In the meantime, copy/paste and edit existing user files to create new users. 
	


mlteld14/changelog
Change Log

1.1 + ablility to choose iphostname or ip address for -UID parsing.
 
1.2 + ability to log both iphostname and ip address
 
1.3 ! fixed bug with passing arguments
    + added sys logging 

1.4 ! fixed a bug that would cause segfault if no hostname was found

MCI/test_anothermci2.py
#!/usr/bin/env python

import re
import sys
import os

ansiColors = {"c0" : "30", "c1" : "31", "c2" : "32", "c3" : "33", "c4" : "34",
               "c5" : "35", "c6" : "36", "c7" : "37", "c8" : "30",
               "ca" : "31", "cb" : "32", "cc" : "33", "cd" : "34",
               "ce" : "35", "cf" : "36", "cg" : "37", "Z0" : "40" ,  "Z1" : "41",
               "Z2" : "42", "Z3" : "43", "Z4" : "44", "Z5" : "45", "Z6" : "46",
               "Z7" : "47"}


def repl(m):
    code = ansiColors[m.group(2)]
    if code[0] != "Z":
      ansitextattrib = "0"
    else: ansitextattrib = "1"
    if code == "c8" or code == "c9" or code == "ca" or code == "cb" or code == "cd" or code == "ce" or code == "cf" or code == "cg":
      ansitextattrib = "1"
    return '\033[' + ansitextattrib + ";" + code + "m"

display = open('sys.infox','r')
for lines in display:
    lines = re.sub(r'(\{)(\w+)(\})', repl , lines)
    print lines.strip('\n')
print ("\033[0m")


bbscore.py
#!/usr/bin/env python3

from datetime import datetime
from passlib.hash import pbkdf2_sha256
import linecache
import getpass
import readline
import sys
import re
import os

# Variable assignments. Will read these in from a config file via linecache and assigned to the correct variable. 
# i.e. bbspath = linecache(bbsconfg,1) and so on. Hopefully this will be able to be created by a GUI to create the file, like BBSCONFIG in CNet. 

ansiColors = {"c0" : "[30m", "c1" : "[31m", "c2" : "[32m", "c3" : "[33m", "c4" : "[34m",
               "c5" : "[35m", "c6" : "[36m", "c7" : "[37m", "c8" : "[1;30m",
               "ca" : "[1;31m", "cb" : "[1;32m", "cc" : "[1;33m", "cd" : "[1;34m",
               "ce" : "[1;35m", "cf" : "[1;36m", "cg" : "[1;37m", "Z0" : "[40m" , "Z1" : "[41m",
               "Z2" : "[42m", "Z3" : "[43m", "Z4" : "[44m", "Z5" : "[45m", "Z6" : "[46m",
               "Z7" : "[47m"}

bbsSetup = 'config.txt'
user = {}
now = datetime.now()
bbsName = linecache.getline(bbsSetup, 1)
bbsName = bbsName.strip('\n')
bbsSysop = linecache.getline(bbsSetup, 2)
bbsSysop = bbsSysop.strip('\n')
bbsCity = linecache.getline(bbsSetup, 3)
bbsCity = bbsCity.strip('\n')
bbsState = linecache.getline(bbsSetup, 4)
bbsState = bbsState.strip('\n')
bbsCountry = linecache.getline(bbsSetup, 5)
bbsCountry = bbsCountry.strip('\n')
bbsFQDN = linecache.getline(bbsSetup, 6)
bbsFQDN = bbsFQDN.strip('\n')
bbsPath = linecache.getline(bbsSetup, 7)
bbsPath = bbsPath.strip('\n')
newsPath = bbsPath + "news/"
sysTextPath = bbsPath + "systext/"
sysDataPath = bbsPath + "sysdata/"
pFilesPath = bbsPath + "pfiles/"
doorsPath = pFilesPath
newsHeaderPath = newsPath + "headers/"
currentcalldate = str(now.year) + str(now.month) + str(now.day)

# Still some clean up that can be done. Also, the user dictionary doesn't get returned back to the whole program. Because it was
# created in a function, it's only good there. So need to work on that. Doing it as a class would probably be the best. Have to read up on how to do that.  
# I'm sure there are ways to slim it down and make it more "pythonic", but getting a rough working framework is the first goal. 
# I'll worry about bells and whistles and fancy coding later on. Next step is to get it to display Welcome and News files, then the Main Menu.


# This function reads the file passed to it and does a regex search for the MCI triggers, and then replaces the MCI trigger with the correct
# ANSI escape codes. 

def repl(m):
    code = ansiColors[m.group(2)]
    return '\033' + code

def mciAnsi(fileName):
  file = open(fileName , 'r')
  for lines in file:
    lines = re.sub(r'(\{)(\w+)(\})', repl , lines)
    print lines.strip('\n')
  print ("\033[0m")
  
# This just displays the sys.start file which is shown before logging on. 
def display_start():
  mciAnsi(sysTextPath + 'sys.start' , 'r') 
  

# This is where the logon info is checked against the user db. Right now the db is just a simple text file. The goal is to use SQLite.
def user_auth(username): 
  pass_tries = 0 
  pass_times = 3
  while pass_tries < pass_times:
    userdb = open(bbsPath + "users/" + username + "/data.txt" , "r") 
    for items in userdb.readlines():
      (key, val) = items.split(":") # Take each line, and split the items between the ":", and save them to key and val
      user[key] = val # Take those pieces of data from above and put them into the dictionary "user" with the corresponding key and value
    print ("\nEnter Password") 
    passwrd = getpass.getpass(": ") 
    if str(passwrd) == user["password"].strip('\n') : #compares password to "password" key in the dictionary. 
      return 
    else: 
      print ("\nIncorrect Password!")
      pass_tries += 1  
  else:
    print ("\nToo many bad login attempts. Try later.")
    sys.exit() # We'll add an option for feedback to Sysop later.

# The logon procedure
def login():
  uname_tries = 0
  uname_times = 3
  while uname_tries < uname_times:
    print("\nEnter \"NEW\" if you have no account.") 
    print("Enter your handle.")
    username = raw_input(": ")
    if len(username) > 0 and username != "new" :
      try:
        userdb = open(bbsPath + "users/" + username + "/data.txt" , "r")
      except IOError:
        print("\nUnknown user.")
        uname_tries += 1
        if uname_tries >= uname_times:
          print("\nToo many unknown user attempts. Contact SysOp!") #Can do a bbstext linecache here.
          sys.exit()
        else:
          pass
      else:
        user_auth(username)
        break
    elif username == 'new':
      new_user_signup()
    elif username =='':
      uname_tries += 1
      if uname_tries >= uname_times:
        print("\nToo many unknown user attempts. Contact SysOp!") #Can do a bbstext linecache here.
        sys.exit()
    else:  
      print("\nToo many bad login attempts. Try again later.") #Can do a bbstext linecache here.
      sys.exit()
  return

# The logoff procedure. Will display the sys.end file. 
def logoff():
  logoffinput = raw_input("\nAre you sure you want to logoff? [y/N]") #We'll use linecache here from BBSTEXT file
  if logoffinput == "y" :
    mciAnsi(sysTextPath + 'sys.end') 
    sys.exit()
  else: return

def new_user_signup(): # New User sign up. Nothing here yet..
  print ("\nNot accepting New Users at this time.")
  sys.exit()

# This grabs a listing of the news files in the news directory
def get_news_items():
  newslist = os.listdir(newsPath)
  return newslist

# This just displays the news files. 
def display_file(item):
  file = open(newsPath + item, 'r') 
  lines = file.readlines()
  del lines[0]
  for i in lines:
    print (i).strip('\n')
  return lines

# This will grab a header for a news file. You'll be able to create multiple headers and assign one to a news item. 
def header_setup(item):
  header = open(newsPath + item , "r")
  header_file = header.readline()
  if header_file[0] == ";":
    header_file = header_file.replace(";" , "")
    header_file = header_file.replace("\n" , "")
    header.close()
    return header_file

# This is to display the news header
def display_header(header_file):
  header = open(newsHeaderPath + header_file , 'r')
  for lines in header:
    print (lines).strip('\n')
  return

# This checks the user's last call date and shows the items that are new. 
def do_news():
  newslist = get_news_items()
  for item in newslist:
    try:
      if int(item.strip('.news')) > int(user['lastcall']):
        header_file = header_setup(item)
        print("\033[2J")
        display_header(header_file)
        display_file(item)
        raw_input('\nPress [ENTER] to continue..\n')
      else:
        pass
    except ValueError:
      pass

def udbase():
  print("\033[2J")
  print("Upload/Download Base Test Menu")
  print("Please choose an option below to test the Zmodem file transfer for this BBS")
  print("1. Recieve our test file\n2. Send us a test file\nQ. Quit")
  udmenu = raw_input("\nChoose (1 or 2 or Quit): ")
  while udmenu != "q" or "Q" :
    if udmenu == "1":
      os.system("sz -b /home/mobbyg/anet/zmodem_test.zip")
      udmenu = "0"
      return
    elif udmenu == "2":
      os.system("cd /home/mobbyg/anet/;rz -bZ")
      udmenu = "0"
      return
    else:
      udmenu ="0"
      return
    
# The main menu for the BBS. 
def main_menu():
  pass
  print ("\nEntering BBS core...")
  while True:
    userinput = raw_input("\nANet BBS >: ") # We'll use linecache here from BBSTEXT file
    if userinput == "o" or userinput == "logoff" :
      logoff()
    elif userinput == "o!" :
      break
    elif userinput == "inf" :
      try:
        mciAnsi(sysTextPath + 'sys.info')
      except IOError:
        print("No system information file found.")
    elif userinput =="u" :
      udbase() 
    else: print ("\nThat Command is not yet active.") #We'll use linecache here from BBSTEXT file
    
#core starts here...    
#display_start() # Function to display the sys.start

# This is pretty much the BBS. Display the start screen, login the user, show the news and then go to the Main Menu. 
# The Main Menu will call the modules for the BBS, like Gfiles, PFiles/Doors and U/D Bases. 
try:
  mciAnsi(sysTextPath + 'sys.start')
except IOError:
  pass
login()
do_news()
try:
  mciAnsi(sysTextPath + 'sys.welcome')
except IOError:
  pass
main_menu() # This is next up. The main menu system


mlteld14/ddtelnetd.c
/* ddtelnetd.c
 *
 * Simple telnet server for Daydream BBS, 100% fast and 100% 8-bit clean.
 * Bo Simonsen <bo@geekworld.dk>

 * Based on utelnetd, see credits below.

 * ---------------------------------------------------------------------------
 * (C) 2000, 2001, 2002, 2003, 2012 by the authors mentioned below
 * ---------------------------------------------------------------------------

 * Artur Bajor, Centrum Informatyki ROW   Joerg Schmitz-Linneweber, Aston GmbH
 * <abajor@cirow.pl>                      <schmitz-linneweber@aston-technologie.de>
 *
 * Vladimir Oleynik                       Robert Schwebel, Pengutronix
 * <dzo@simtreas.ru>                      <r.schwebel@pengutronix.de>
 *
 * Bjorn Wesen, Axis Communications AB    Sepherosa Ziehau
 * <bjornw@axis.com>                      <sepherosa@myrealbox.com>
 *
 * Frank Linhares
 * <frank@bbs-scene.org>
 *
 * This file is distributed under the GNU General Public License (GPL),
 * please see the file LICENSE for further information.
 *
 * Enthral changes by Micheal Griffin
 * Mystic changes by Frank Linhares
 *
 * Mystic BBS Revision 1.4
 * September 19 2012
 *
 * Change Log
 *
 * + ablility to choose iphostname or ip address for -UID parsing.
 *
 * + ability to log both iphostname and ip address
 * 
 * ! fixed bug with passing arguments
 *
 * + added sys logging
 *
 * ! fixed a segfault where ddtelnetd would crash if no hostname was found
 **/

#include <sys/time.h>
#include <sys/wait.h>
#include <sys/stat.h>
#include <sys/ioctl.h>
#include <sys/types.h>

#include <unistd.h>
#include <pty.h>

#include <cstdio>
#include <cstdlib>
#include <cerrno>
#include <csignal>
#include <cctype>
#include <cstring>

#include <fcntl.h>
#include <syslog.h>

#include <arpa/telnet.h>
#include <arpa/inet.h>
#include <net/if.h>

#include <sys/socket.h>
#include <netinet/in.h>
#include <netdb.h>


#define BUFSIZE 4096

#define OUTFD 1
#define INFD 0

#define FALSE 0
#define TRUE 1

static char *argv_init[] = {NULL, NULL, NULL, NULL, NULL};

void show_usage(void)
{
	printf("Usage: telnetd [-l loginprogram] [-u user]\n");
	printf("\n");
	printf("   -l loginprogram  program started by the server\n");
	printf("   -u user  logins user without password auth\n");
	printf("\n");
	exit(1);
}

int send_iac(unsigned char command, int option);

int
remove_iacs(unsigned char *in, int len, unsigned char* out, int* out_count) {

	unsigned char *end = in + len;
	unsigned char* orig_in = in;
	static int ignore = FALSE;

	while (in < end) {
		if (*in != IAC) {
			if(ignore == FALSE) {
				*out = *in; in++; out++;
				*out_count = *out_count + 1;
			}
			else {
				in++;
			}
		}
		else {
			if((in+1) == end) {
				*orig_in = IAC;
				return 1;
			}

			if(ignore && *(in+1) != SE) {
				send_iac(*(in+1), -1);
				in += 2;
				continue;
			}

			switch(*(in+1)) {
			case IAC:
				*out = IAC; ++out;
				*out_count = *out_count + 1;
				in += 2;
				break;
			case WILL:
			case WONT:
			case DO:
			case DONT:
				if((in+2) == end) {
					*orig_in = IAC;
					*(orig_in+1) = *(end-1);
					return 2;
				}
				in += 3;
				break;
			case SB:
				ignore = TRUE;
				syslog(LOG_INFO, "negotiation began");
				in += 2;
				break;
			case SE:
				ignore = FALSE;
				syslog(LOG_INFO, "negotiation completed");
				in += 2;
				break;
			case NOP:
			case DM:
			case BREAK:
			case IP:
			case AO:
			case AYT:
			case GA:
				in += 2;
				break;
			case EC:
				in += 2;
				*out = '\b';
				++out;
				*out_count = *out_count + 1;
				break;
			default:
				in += 2;
				break;
			}
		}

	}

	return 0;
}

int send_iac(unsigned char command, int option)
{
	char buf[3];
	int w;

	buf[0] = IAC;
	buf[1] = command;
	buf[2] = option;

	w = write(OUTFD, buf, option == -1 ? 2 : 3);

	return w;
}

int main(int argc, char **argv)
{
	int new_offset = 0;
	int pid;
	int shell_pid;
	int ptyfd;

	unsigned char buf[BUFSIZE];
	unsigned char tmpbuf[BUFSIZE];
	char loginpath[256];
	char host_string[256];
	char ip_string[256];

	char* user = NULL;
	char* login = NULL;

	struct sockaddr_in from;
	socklen_t fromlen;
	const char *ip;
	const char *hostname;
	struct hostent *hp;


	for (;;) {
		int c;
		c = getopt( argc, argv, "l:u:");
		if (c == EOF) break;
		switch (c) {
			case 'u':
				user = strdup(optarg);
				break;
			case 'l':
				login = strdup(optarg);
				break;
			default:
				printf("%c\n", c);
				show_usage();
				exit(1);
		}
	}

	if(login) {
		strcpy(loginpath, login);
		free(login);
		login = NULL;
	} else {
		strcpy(loginpath, "/bin/login");
	}

	argv_init[0] = strdup(loginpath);

	/* Michael Griffin Get Host Address
        From Remote Connection, Pass Like in.telnetd does.

	Frank Linhares - Pass -IP and -HOST before hostname for Mystic */

	fromlen = sizeof (from);
	if (getpeername(0, (struct sockaddr *)&from, &fromlen) < 0) {
		perror("Error: getpeername");
	}

	/* Map the IP address to a host name. */
	hp = gethostbyaddr((char *)&from.sin_addr, sizeof(struct in_addr), from.sin_family);
	if (hp)
	{
    	/* Got host name. */
		hostname = hp->h_name;
	}
	else
    {
      /* Host name not found. pass the ip address as hostname. */
      hostname = inet_ntoa(from.sin_addr);
    }

	/* get users current ip address */

	ip = inet_ntoa(from.sin_addr); /// get ip address
                               
	/* Mystic BBS - Pass -HOST to mystic. */
	
	sprintf(ip_string,"-IP%s",strdup(ip));
	argv_init[1] = ip_string;

	/* Mystic BBS - Pass -IP to mystic. */
	
	sprintf(host_string,"-HOST%s",strdup(hostname));
	argv_init[2] = host_string;
	
	syslog(LOG_INFO, "incoming connection from %s / %s", strdup(ip), strdup(hostname));

    /* These options are taken from maximus ipcomm.c (max 3.03b unix) */

	if(send_iac(DONT, TELOPT_OLD_ENVIRON) < 0) {
	  exit(1);
	}
	if(send_iac(DO, TELOPT_SGA) < 0) {
	  exit(1);
	}
	if(send_iac(WILL, TELOPT_ECHO) < 0) {
	  exit(1);
	}
	if(send_iac(WILL, TELOPT_SGA) < 0) {
	  exit(1);
	}
	if(send_iac(WONT, TELOPT_NAWS) < 0) {
	  exit(1);
	}
	if(send_iac(WILL, TELOPT_BINARY) < 0) {
	  exit(1);
	}
	if(send_iac(DO, TELOPT_BINARY) < 0) {
	  exit(1);
	}

    openlog("ddtelnetd", LOG_PID, LOG_SYSLOG);

	pid = forkpty(&ptyfd, NULL, NULL, NULL);

	if (pid == 0) {
		setsid();
		tcsetpgrp(0, getpid());

        /* exec shell, with correct argv and env */
		execv(loginpath, argv_init);
		exit(1);
	}
	else if(pid == -1) {
	    closelog();
	    exit(1);
	}
	else {
	    shell_pid = pid;
	}

	do {
		int selret;
		fd_set rdfdset;

		FD_ZERO(&rdfdset);
		FD_SET(ptyfd, &rdfdset);
		FD_SET(INFD, &rdfdset);

		selret = select(ptyfd + 1, &rdfdset, NULL, 0, 0);
		
		if (selret <= 0)
			break;

		if (FD_ISSET(ptyfd, &rdfdset)) {

			unsigned char* ptr1, *ptr2;	
			int iacs = 0;
    			int r;
    		
    			r = read(ptyfd, tmpbuf, BUFSIZE / 2);
	    		
			if (r <= 0) {
				break;
			}

			ptr1 = tmpbuf;
			ptr2 = buf;

			while(ptr1 != tmpbuf + r && ptr2 != buf + BUFSIZE) {
				if(*ptr1 == IAC) {
					*ptr2 = IAC; ptr2++;
					*ptr2 = IAC; ptr2++;
					ptr1++; iacs++;
				}
				else {
					*ptr2 = *ptr1;
					ptr1++; ptr2++;
				}
			}
			
			if(write(OUTFD, buf, r + iacs) != (r + iacs)) {
			   break;
			}
		}

		if (FD_ISSET(0, &rdfdset)) {
			int out_count = 0;
			int r;

			r = read(INFD, buf + new_offset, BUFSIZE - new_offset);
				
			if (r <= 0) {
				break;
			}

			new_offset = remove_iacs(buf, r + new_offset, tmpbuf, &out_count);
			
			if(write(ptyfd, tmpbuf, out_count) != out_count) {
			   break;
			}
		}

	} while (1);

	close(ptyfd);

	syslog(LOG_INFO, "closed connection");
	closelog();
	
	_exit(0);
	
	return 1;
}


MCI/copy_anothermci2.py
#!/usr/bin/env python

import re
import sys
import os

def mciInterp(m):
    ansiBkGColor = "0"
    ansiFgColor = "c"
    ansiBold = "0"
    mciTemp = m.group(2)
    mciTemp = mciTemp.split(",")
    mciCMD1 = mciTemp[0]
    try:
        stringCMD2 = mciTemp[1]
    except IndexError:
        stringCMD2 = "None"
        pass

    if mciCMD1[0] != "Z":
        pass
    elif mciCMD1[0] == "Z":
        if mciCMD1[1] == "0":
            ansiBkGColor == "40" # Black
        elif mciCMD1[1] == "1":
            ansiBkGColor == "41" # Red
        elif mciCMD1[1] == "2":
            ansiBkGColor == "42" # Green
        elif mciCMD1[1] == "3":
            ansiBkGColor == "43" # Yellow
        elif mciCMD1[1] == "4":
            ansiBkGColor == "44" # Blue
        elif mciCMD1[1] == "5":
            ansiBkGColor == "45" # Purple
        elif mciCMD1[1] == "6":
            ansiBkGColor == "46" # Cyan
        elif mciCMD1[1] == "7":
            ansiBkGColor == "47" # White

    if mciCMD1[0] != "C":
        pass
    elif mciCMD1[0] == "C":
        if mciCMD1[1] == "0":
            ansiFgColor == "30" # Black
        elif mciCMD1[1] == "1":
            ansiFgColor == "31" # Red
        elif mciCMD1[1] == "2":
            ansiFgColor == "32" # Green
        elif mciCMD1[1] == "3":
            ansiFgColor == "33" # Yellow
        elif mciCMD1[1] == "4":
            ansiFgColor == "34" # Blue
        elif mciCMD1[1] == "5":
            ansiFgColor == "35" # Purple
        elif mciCMD1[1] == "6":
            ansiFgColor == "36" # Cyan
        elif mciCMD1[1] == "7":
            ansiFgColor == "37" # White
        elif mciCMD1[1] == "8":
            ansiFgColor == "30"
            ansiBold == "1"     #From here down will turn on bold for the respected color
        elif mciCMD1[1] == "9":
            ansiFgColor == "31"
            ansiBold == "1"
        elif mciCMD1[1] == "a":
            ansiFgColor == "32"
            ansiBold == "1"
        elif mciCMD1[1] == "b":
            ansiFgColor == "33"
            ansiBold == "1"
        elif mciCMD1[1] == "c":
            ansiFgColor == "34"
            ansiBold == "1"
        elif mciCMD1[1] == "d":
            ansiFgColor == "35"
            ansiBold == "1"
        elif mciCMD1[1] == "e":
            ansiFgColor == "36"
            ansiBold == "1"
        elif mciCMD1[1] == "f":
            ansiFgColor == "37"
            ansiBold == "1"

    return '\033[' + ansiBold + ";" + ansiBkGColor + ";" + ansiFgColor + "m"
    #return ansi


display = open('sys.infox','r')
for lines in display:
    lines = re.sub(r'(\{)(\w+)(\})', mciInterp, lines)
    print lines.strip('\n')
